docs/reference/api/platform/NonCopyable.md

```diff
@@ -1,5 +1,11 @@
 ## NonCopyable
-This API tags a class as not supporting copy operations. It creates a compile-time error if you copy the object.
+
+The NonCopyable class prevents objects of a class from supporting copy operations. You can easily identify it from its class declaration. It creates a compile-time error if you copy the object. Inheriting from this class results in autogeneration of private copy construction and assignment operations, which are not accessible in derived classes.
```

Comment:
` assignment operations` shall be `copy assignment operations` 

It might be worth mentioning: when to use it - usually when a class own a resource ?